### DISCUSSION TITLE
How to generate a turbine model

### DISCUSSION THREAD

**Discussion by ArthurFung1998** [NatLabRockies/floris#462] [Q&A] [answered]
The models defined in FLORIS is just THREE types, which are nrel 5MW,  iea 10MW and iea 15MW.
How could I gernerate a new turbine model like 4MW or less?
I found that the most important parameter is the power-thrust-windspeed table, I try to make one from a well-known turbine simulation software-Bladed, but the simulation result shows a huge difference between Floris and Bladed.
BTW, if I wanna generate a turbine model from actual SCADA data or LOG data, what kind of data will be needed?

**Comment by paulf81** [comment, marked as answer]
I'm sorry for missing this question and only answering now.  Typically we use a tool such as openfast or ccblade to generate cp/ct tables if these are not provided by a turbine spec sheet.  

Note in our examples we assume cp is electrical cp (so including generator efficiency into cp and not using the separate generator_efficiency term)

It's possible to back-out Cp from SCADA data but difficult for Ct because thrust is not often measured.  In these cases it could be good to scale a known model (such as one of the IEA references)
